Scripts/SAPSetupAssistantValidationScripts/SetupAssistantSAPValidateGateway.ps1

```diff
@@ -0,0 +1,73 @@
+<#
+.SYNOPSIS
+    Checks for On-permises data gateway installation.
+.DESCRIPTION
+    This script attempts to
+    1. Check if the On-permises data gateway is installed and validates the
+    if the version is greater than the minumum version requirement
+    2. CheckSapConnector
+    3. CheckVisualCPlusPlus
+#>
+
+$minimumVersion = "3000.150.11" # November 2022
+$serviceName = "PBIEgwService"
+$service = Get-Service $serviceName -ErrorAction SilentlyContinue
+
+if ($service) {
+    $servicePath = (Get-WmiObject Win32_Service | Where-Object { $_.Name -eq $serviceName }).PathName
+    $exePath = $servicePath.Trim('"')
+    $versionInfo = (Get-Command $exePath).FileVersionInfo
+    $versionNumber = $versionInfo.ProductVersion
+
+    if ([version]$versionNumber -ge [version]$minimumVersion) {
+        Write-Host -NoNewLine "{'step': 'CheckGatewayVersion', 'status': 'Success', 'message': 'The on-premises data gateway is installed and version ($versionNumber) is greater than or equal to ($minimumVersion).'},"
+    }
+    else {
+        Write-Host -NoNewLine "{'step': 'CheckGatewayVersion', 'status': 'Error', 'message': 'Your on-premises data gateway doesn't meet the minimum version, please upgrade by downloading the latest version here: https://go.microsoft.com/fwlink/?linkid=2240537.'},"
+    }
+    }
+else {
+    Write-Host -NoNewLine "{'step': 'CheckGatewayVersion', 'status': 'Error', 'message': 'The on-premises data gateway is not installed. Download it here: https://go.microsoft.com/fwlink/?linkid=2240537.'},"
+}
+
+    # .NET SDK for SAP
+    $currentLocation = Get-Location
+    $location = "C:\Windows\Microsoft.NET\assembly"
+    $ncoAssemblies = Get-ChildItem -Path $location -Filter sapnco.dll -Recurse | Select-Object -ExpandProperty VersionInfo
+    $location = "C:\Program Files\SAP\SAP_DotNetConnector3_Net40_x64"
+    $ncoAssemblies += Get-ChildItem -Path $location -Filter sapnco.dll -Recurse | Select-Object -ExpandProperty VersionInfo
+    $counter = 0
+
+    foreach ($assembly in $ncoAssemblies) {
```

Comment:
Might want to add a check before the loop for the condition that $ncoAssemblies does not contain assembly items and handle that gracefully.

Reply:
With Powershell 3 the foreach statement does not iterate over $null, even if there are no assembly items foreach won't be executed. Do we still need to handle this ?